Added a reversed intake function

Archonic944 opened the pull request:
For unjamming the note. Ideally bound to a trigger at a high threshold.

Archonic944 committed:
Added a reversed intake function

BlueZeeKing committed:
Fix formatting

BlueZeeKing commented on src/main/java/frc/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -19,14 +19,23 @@ public IntakeSubsystem() {
+        () -> motor.set(IntakeConstants.SPEED * (inverted ? -1 : 1)), () -> motor.set(0), this);
+  }
+
   private Command runIntake() {
```
We don't need to keep this override.

BlueZeeKing replied:
It's also unused I believe

BlueZeeKing commented on src/main/java/frc/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -19,14 +19,23 @@ public IntakeSubsystem() {
+    return intake(shooter, false);
+  }
+
+  public Command intake(ShootSubsystem shooter, boolean reversed) {
```
This performs a full intake including ensuring the shooter is properly configured, and the note gets pulled into the correct position into the bot. Running the intake in reverse during this process would result in a command that never exits because the note never enters the shooter.

Archonic944 replied:
Got it. Should I just make it run for a set time, then? Another option is to make it run while the driver is holding the trigger. Also, could you direct me to the line that interrupts when a note is detected?

+we should probably be able to manually kill intake

BlueZeeKing replied:
Just run for a set amount of time for now. I want to do some more physical testing of ways the note can get stuck before we make a final decision

BlueZeeKing reviewed (changes requested):
The full `intake` command shouldn't run the intake in reverse, as the command will run forever while waiting for a note. Instead, another command that ejects the note should be made public.

- [x] Create eject note command
- [x] Remove extra `runIntake` override

Archonic944 commented:
I found it.
 `return Commands.waitUntil(
      () -> Math.abs(intermediateEncoder.getPosition()) > IntermediateConstants.TOLERANCE);`
But shouldn't we use the tiny little switch as well?

Archonic944 committed:
Add note ejection

for real, for real, for real this time

Archonic944 committed:
Merge branch 'inverted' of https://github.com/Ocebots/ocebot-code-2024 into inverted

Archonic944 committed:
Intake timeout ==> note ejection

Archonic944 committed:
Intake timeout ==> note ejection

Archonic944 committed:
Merge branch 'inverted' of https://github.com/Ocebots/ocebot-code-2024 into inverted

BlueZeeKing reviewed (changes requested):
I would rather the intake success field be within the `IntakeSubsystem` instead of the `ShootSubsystem`

Archonic944 committed:
FINE

but I'll have you know we just made it significantly slower because of how runnables work
actually maybe not idk

BlueZeeKing committed:
Move intakeSuccess completely into the intake
